src/app/rpmostree-builtin-status.cxx

```diff
@@ -639,6 +639,11 @@ print_one_deployment (RPMOSTreeSysroot *sysroot_proxy,
             g_print ("%s", canonrefspec);
           }
           break;
+        case RPMOSTREE_REFSPEC_TYPE_CONTAINER:
+          {
+            g_print ("%s", canonrefspec); // Note: this is currently unreachable because `origin_refspec` will always be `ostree://`
```

Comment:
This is because what we store in `origin_refspec` is always just the `refspecdata`/`remainder` side of the `refspec`, and for compatibility, if we can't identify a refspectype via the prefix, we default to `ostree://`.
I think we many need to change this. I think this is an example of the awkwardness of storing a refspec/imgref as a string and naming the field refspec. Throughout the code, I've noticed inconsistencies in what `refspec` refers to.

Reply:
If it's the same thing, we can combine the `case` statements right?

Reply:
Should it be the same thing though? For example, would there be some special fields we'd like to display in image references vs refspecs? 
Related to this, should the "canonical" form be `ostree://fedora:fedora/x86_65/coreos/stable`/`container://registry:quay.io/fcos`? This way, we always clearly differentiate between image refs and refspecs. I think this might be useful when we start considering `deploy` (via version or commit) and `upgrade`. Otherwise, we'll rely solely on "deducing the refspectype" as discussed in https://github.com/coreos/rpm-ostree/issues/2909#issuecomment-867998260.

Reply:
Yeah...I think I debated changing it to be `ostree://` by default when we added `rojig://` but the weird thing then is the string becomes something you can't copy-paste into the `ostree` CLI, i.e. you can't do `ostree pull ostree://fedora:fedora/x86_65/coreos/stable`.  But, we could make that work in ostree upstream!

Reply:
hmm OK this was a bit more involved than I thought.
I thought it would be a good idea to make "refspec" mean "refspectype" + "refspecdata" throughout the code, but it looks like in the vast majority of places "refspec" is assumed to be what I considered "refspecdata", and changing hundreds of instances of "refspec" in the existing code doesn't sound too feasible or smart. I now see how elegant it would be if we could always reliably _classify_ the refspec and retrieve its "refspectype" just from the "refspecdata" portion.